python/cuml/common/base.pyx

```diff
@@ -282,6 +282,51 @@ class Base:
             else:
                 raise AttributeError
 
+    def _set_base_attributes(self,
+                             X,
+                             output_type=False,
+                             y=None,
```

Comment:
this parameter could have a better name, something like `target_dtype=None`, and in general the api will be easier to read if the parameters match the attributes that they get assigned to.

Reply:
Same as comment below. This set on X, instead of the `self._set_target_type(y)` which is used only by certain models.

Reply:
Sorry for all the nitpickiness! I think this is the last detail, but perhaps this `y` could be `target_dtype` no? Since  `_set_output_type` only sets the `self.output_type`, while this sets `self.target_dtype`. 

Also could you remove the docstring of the other `_set_bla` methods? That will help to avoid confusion for new devs

Reply:
Sounds good, making the changes now!